packages/react-native-web/src/exports/Image/index.js

```diff
@@ -211,24 +211,18 @@ const BaseImage: ImageComponent = React.forwardRef((props, ref) => {
     }
   }
 
-  const [state, updateState] = React.useState(() => {
-    const uri = resolveAssetUri(source);
-    if (uri != null) {
-      const isLoaded = ImageLoader.has(uri);
-      if (isLoaded) {
-        return LOADED;
-      }
-    }
-    return IDLE;
-  });
-
+  const [state, updateState] = React.useState(IDLE);
```

Comment:
This is a state initializer, it runs only once, because of that the component does not behave correctly when we switch `source`, cases like 
- component is mounted with `null` source then changed to actual source that was already loaded - state was already initialized and is not updated to LOADED
- component changes source dynamically to some other source that was loaded - same problem as above

This causes flickers on some occasions